### PR TITLE
Feature/home screen

### PR DESCRIPTION
**Pull request by Kovler10**
Added home screen like in figma

- custom component for bottom drawer
- extracted menu button from drawer component
- added home screen to side menu and routes
- moved rides mock into hook as well

### CODE REVIEW

File: apps/ui/src/Components/bottom-drawer/styles.ts
Comment (DarNachmani) [inline apps/ui/src/Components/bottom-drawer/styles.ts:1]: Why do you need these when you have MUI SwipeableDrawer?
Like in the following example: https://mui.com/material-ui/react-drawer/#swipeable-edge

File: apps/ui/src/Components/bottom-drawer/bottom-drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { Box } from '@mui/material';
+import React, { useState, useEffect, PropsWithChildren } from 'react';
+
+import {
```
Comment (DarNachmani) [inline apps/ui/src/Components/bottom-drawer/bottom-drawer.tsx:4]: Look at the comment in the styles.ts file

File: apps/ui/src/Components/community-list/community-list.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import { RideCard } from '../ride-card/ride-card';
+
+dayjs.extend(relativeTime);
+
+type CommunityWithRides = Community & { rides: any[] };
+
+interface CommunityListProps {
+  communities: CommunityWithRides[];
+}
```
Comment (DarNachmani) [inline apps/ui/src/Components/community-list/community-list.tsx:17]: Put these in a types file

File: apps/ui/src/Components/community-list/community-list.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+const CommunityList: React.FC<CommunityListProps> = (props) => {
+  const { communities } = props;
+
+  return (
+    <>
+      {communities.map((community, index) => (
+        <div key={index}>
```
Comment (DarNachmani) [inline apps/ui/src/Components/community-list/community-list.tsx:25]: Use Box of MUI instead of div

File: apps/ui/src/Pages/SignIn/SignIn.tsx [hunk 4]
```diff
@@ -1,26 +1,21 @@
...
-import { isEmpty } from 'lodash';
+import { PasswordField } from '../../Components/Signing/Fields/PasswordField';
 
 interface SignInProps {
-  setMenuVisible: (value: boolean) => void;
+  setMenuVisible: Dispatch<SetStateAction<boolean>>;
 }
 
-const SignIn = ({ setMenuVisible }: SignInProps) => {
+const SignIn: React.FC<SignInProps> = ({ setMenuVisible }) => {
```
Comment (DarNachmani) [inline apps/ui/src/Pages/SignIn/SignIn.tsx:18]: Why do we need `: React.FC<SignInProps>`?

File: apps/ui/src/Pages/SignIn/SignIn.tsx [hunk 5]
```diff
@@ -60,11 +59,11 @@ const SignIn = ({ setMenuVisible }: SignInProps) => {
       email: data.get('email'),
       password: data.get('password'),
     };
-    console.log({userSignIn});
-    
-    if(isSubmitEnabled) {
+    console.log({ userSignIn });
```
Comment (DarNachmani) [inline apps/ui/src/Pages/SignIn/SignIn.tsx:62]: Remove

File: apps/ui/src/Pages/SignIn/SignIn.tsx [hunk 6]
```diff
@@ -33,23 +28,27 @@ const SignIn = ({ setMenuVisible }: SignInProps) => {
   const [isSubmitEnabled, setIsSubmitEnabled] = useState(false);
 
   useEffect(() => {
```
Comment (DarNachmani) [inline apps/ui/src/Pages/SignIn/SignIn.tsx:30]: You don't need this useEffect, you can check for errors in the onChange function

File: apps/ui/src/Pages/home-page/home-page.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+} from '../../hooks/Communities/useGetAllCommunities';
+
+const HomePage: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState<'communities' | 'rides'>(
+    'communities',
+  );
+  const [isProfileOpen, setIsProfileOpen] = useState(false);
+
+  const communities = useMemo(() => {
+    const base = useGetAllCommunities();
```
Comment (DarNachmani) [inline apps/ui/src/Pages/home-page/home-page.tsx:24]: Rename to `baseCommunities` like `baseRides `

File: apps/ui/src/Pages/home-page/home-page.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    const baseRides = useGetAllRides();
+
+    const groupedRides = groupBy(baseRides, 'communityName');
+
+    return base.map((com) => ({ ...com, rides: groupedRides[com.name] ?? [] }));
```
Comment (DarNachmani) [inline apps/ui/src/Pages/home-page/home-page.tsx:29]: It is better to call it community and not com - easier to understand

File: apps/ui/src/Pages/home-page/styles.ts
Comment (DarNachmani) [inline apps/ui/src/Pages/home-page/styles.ts:1]: Don't use styled components when we have everything we need in MUI

File: apps/ui/src/Pages/home-page/home-page.tsx
Comment (DarNachmani) [inline apps/ui/src/Pages/home-page/home-page.tsx:1]: Rename to HomePage - we should have the same naming pattern to all files

File: apps/ui/src/Components/ride-card/ride-card.tsx
Comment (DarNachmani) [inline apps/ui/src/Components/ride-card/ride-card.tsx:1]: Rename to RideCard- we should have the same naming pattern to all files

File: apps/ui/src/Components/community-list/community-list.tsx
Comment (DarNachmani) [inline apps/ui/src/Components/community-list/community-list.tsx:1]: Rename to CommunityList- we should have the same naming pattern to all files

File: apps/ui/src/Components/bottom-drawer/bottom-drawer.tsx
Comment (DarNachmani) [inline apps/ui/src/Components/bottom-drawer/bottom-drawer.tsx:1]: Rename to BottomDrawer - we should have the same naming pattern to all files

File: apps/ui/src/Components/Menu/menu.tsx
Comment (DarNachmani) [inline apps/ui/src/Components/Menu/menu.tsx:1]: Rename to Menu - we should have the same naming pattern to all files

**Review by DarNachmani — COMMENTED** [review]
Nice, left some comments.
We should pay attention to the naming pattern for component files - should be the same in all components

File: apps/ui/src/Components/bottom-drawer/styles.ts
Comment (Kovler10) [reply apps/ui/src/Components/bottom-drawer/styles.ts:1]: it doesn't work properly on desktop and doesn't let you interact with the shown elements unless open

File: apps/ui/src/Components/community-list/community-list.tsx (hunk 2 above)
Comment (Kovler10) [reply apps/ui/src/Components/community-list/community-list.tsx:17]: I left the props interface but moved the other type to types file

File: apps/ui/src/Components/bottom-drawer/styles.ts
Comment (DarNachmani) [reply apps/ui/src/Components/bottom-drawer/styles.ts:1]: What kind of interaction do you need?
And this is an app for phones, isn't it? I find it strange that it is not good enough .
You can try adding something like the following Box element to the SwipeableDrawer to make it close and open on click:
`<Box
          onClick={toggleDrawer(!isOpen)} // Toggle on click
          onKeyDown={toggleDrawer(!isOpen)}
>`
It is bad practice to create these custom elements when we are using MUI

**Review by DarNachmani — COMMENTED** [review]
Nice, one comment :)

File: apps/ui/src/Pages/Map/index.tsx [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
 
 // Define start point and destinations
-const startPoint: [number, number] = [34.781769, 32.079444]; // Example start point
-const destinations = [
-  // Example destinations
-  { longitude: 34.773513, latitude: 32.095531 },
-  { longitude: 34.801461, latitude: 32.113314 },
-  { longitude: 34.751869, latitude: 32.050425 },
-];
+// const startPoint: [number, number] = [34.781769, 32.079444]; // Example start point
```
Comment (DarNachmani) [inline apps/ui/src/Pages/Map/index.tsx:5]: Why not remove this?

File: apps/ui/src/Pages/Map/index.tsx [hunk 10]
```diff
@@ -24,11 +24,7 @@ const MapPage: React.FC = () => {
 
   return (
     <div style={mapContainerStyle} className="full-viewport">
-      <Map
```
Comment (DarNachmani) [inline apps/ui/src/Pages/Map/index.tsx:27]: Why?
And can you change the div to MUI Box?

**Review by DarNachmani — COMMENTED** [review]
Nice, small comments

**Review by DarNachmani — APPROVED** [review]

**Pull request merged**